djconnectwise/sync.py

```diff
@@ -2906,10 +2909,61 @@ def update(self, record, changed_fields, **kwargs):
             api_public_key=kwargs.get('api_public_key'),
             api_private_key=kwargs.get('api_private_key')
         )
-        # convert the fields to the format that the API expects
-        api_fields = self._convert_fields_to_api_format(changed_fields)
 
-        updated_record = client.update_ticket(record, api_fields)
+        # Handle dependency error: The API throws an error if a ticket has
+        # a predecessor. To work around this, we first remove the predecessor,
+        # update the required fields, and then re-add the predecessor after
+        # the update.
+        predecessor_reset = True
+        if record.ticket_predecessor_id and (
+            changed_fields.get('estimated_start_date') or
+            changed_fields.get('required_date_utc')
+        ):
+
+            predecessor_reset_fields = {
+                'ticket_predecessor': None,
+                'required_date_utc': None,
```

Comment:
> Does the required_date_utc NEED to be set as none for this as well? I don't think it does, and that creates the problem of it also removing this and never adding it back. Check to make sure we don't have to null this field.

Yes, I have added it on purpose if we don't remove the `required_date_utc` or `estimated_start_date` the same dependency error occurs.

Reply:
> Oh, also this appears to happen when even the start date is updated. So if a user updates a start date, doesn't this erase the required date and then not add it back?

Yes, if a ticket has predecessor and user change either start or end date we need to set both to null and then set the date. Otherwise the dependecy error occurs